### ISSUE TITLE
Camera definitions for use in toolbox

### ISSUE THREAD

**Issue by woodbe** [biometricITC/cPP-toolboxes#25]
Originally when looking at image-based modalities for mobile devices, we had ended up with 3 levels of camera, 2 from the mobile device and then a higher-end DSLR. Generally there defined this way:

- Selfie camera - image had to be taken with something approximately at the resolution of the front-facing camera, which today is commonly around 8MP (though used to be 4MP or much lower)
- Rear-facing camera - image had to be taken with something approximately at the resolution of the main camera, commonly in the 12-16MP range.
- DSLR camera - image to be taken on something closer to 40MP range

Today it isn't as clear that there needs to be three levels, or how they should be divided. While this may be specific to the modality, for example vein images may only have specific types of cameras in use, it isn't clear how applicable this may be across the board or to just face/eye.

In some ways this is a continuation of #7 in terms of debating the types of tests that should be handled, but here the point is what level of camera equipment should be used.

The initial question here is to we need the Selfie camera to be used in testing, or is that considered too low at this point for our focus. Similarly, with some phones coming with very high resolution cameras in them, does it matter if the high resolution pictures are taken with a DSLR camera or only that they are of a level of resolution?

**Comment by n-kai** [comment]
The toolboxes have to conform to the CC/CEM and AVA_VAN.1-5 requires evaluator to look for publicly available information (e.g. research papers) to devise penetration tests. The following is my suggestions based on such information for each biometric modality.

- Iris 
According to *1), _iris image should be captured with high resolution infrared camera in the way that iris diameter must have size of at least 250–300 dots on the paper printed with resolution of at least 600 dpi, which means the image should be captured either from a very short distance or using high-resolution telephoto lens; eyes should be opened enough with the sight directed to the camera; iris image should not be blurred and under- or overilluminated._
So primary focus of iris PAIs are those ones created with the DSLR camera with NIR light (such camera can be rented). However, if the evaluator can't succeed the attack with those PAIs, the evaluator should move to other types of PAIs (e.g. DSLR camera with visible light and so on) to make sure that there is no PAI that reliably break the TOE but the evaluator should not spend lots of times with such PAIs because there is no research papers that recommend to use such PAIs, but the evaluator can skip "Selfie camera" because any iris recognition algorithm require a clear iris image to create iris code. 
*1) "Iris Anti-Spoofing Solution for Mobile Biometric Applications", G. Odinokikh et al

- 2D face
There are several PAD algorithm and most of them are weak against the high resolution PAIs but some of them are weak against the low resolution PAIs. The evaluator doesn't know which PAD algorithm is implemented in the TOE so the evaluator should test all type of PAIs. But the evaluator can skip "Selfie camera" because it's better to change environment condition (e.g. ambient light) rather than using low resolution camera.

- vein
vein is the same as the iris and the evaluator needs to capture the vein pattern with NIR camera. 

I will update the toolbox overview if all agree with my suggestions

**Comment by woodbe** [comment]
My only question about the eye is that if someone is trying to use a visible light pattern of the eye, instead of NIR, then saying the only tests should be on NIR seems to preclude the possibility of other options. I would be fine with saying "choose based on sensor type" here.

I'm not saying I know of any specific sensor using visible light for the eye (i.e. visible iris pattern, which would still need similar resolution), I just didn't want to preclude it, which is why the test which had originally been written as "iris" were changed to "eye" and the tests broadened to cover both visible and NIR light.

Other than that, I'm perfectly fine with this.

**Comment by n-kai** [comment]
I agree with you and I am updating toolbox overview considering your comment that will be done tomorrow.

**Comment by woodbe** [comment]
I merged the other changes so you can work from the current version in a new PR.

**Comment by woodbe** [comment]
For cameras, move to 2 types/classes, not 3, but leave the cut off between them to be determined based on supporting research papers. Generally recommendation is probably at least 20MP camera for the high end. Also, no longer define the camera type (i.e. mobile vs DSLR) unless specifically called out in research paper.

Also note that if there is a conflict between recommendation and a research paper, the research paper takes precedence in terms of equipment.

**Comment by woodbe** [comment]
Proposed camera resolutions:

Type 1 = roughly the resolution of the biometric sensor (within 10%)
Type 2 = 3X (or higher) the resolution of the biometric sensor

@n-kai will update face and @woodbe will update eye with these types

**Comment by woodbe** [comment]
@woodbe will create a pull request for the research paper conflict into the toolbox overview

**Comment by n-kai** [comment]
I looked at latest research papers for 2D face anti-spoofing and the following is my recommendation for 2D face toolbox:

a) We should use at least two different types of camera (i.e. smartphone and dSLR) to create widely different quality (e.g. texture) of PAI species (e.g. printed photo). It may depends on face PAD algorithm but the algorithm tends to overfit to particular type of PAI species (e.g. ones created by smartphone) that the algorithm is trained and it is weak to those PAI species that the algorithm has never seen before (e.g. ones created by the latest dSLR) (See *1))

b) We should create the highest quality face PAI species with the latest product within reasonable cost (e.g. Printed photo created the latest dSLR and printer with highest resolution) because the technology is changing rapidly and the PAD algorithm may not see such PAIs before.

So the face toolbox should define two types of camera resolution.

Type1 = the same quality of the biometric sensor the TOE embeds as @woodbe suggested.
Many users upload their picture to SNS using this Type1 resolution.

Type 2 = the highest resolution of the dSLR that is released within one or two years (e.g. 50MP). A few users upload their picture using Type2 resolution but it is expected number of such user will increase soon.

During the AVA_VAN.1 test, evaluator should focus on Type 2 PAI species changing other parameters (e.g. change inkjet printer to laser).

I will reduce the number of tests in 2D face toolbox as I suggested if other members agree.

*1) https://arxiv.org/pdf/1901.05633.pdf

**Comment by woodbe** [comment]
@n-kai I get the general definition of the two types, but how would this fit in with the camera type? For example, would Type 1 be from a mobile device (any device with a similar camera resolution since the preference would be not to use the device under test) and then Type 2 is the dSLR? 

I'm OK with breaking it down to two different devices this way, but it isn't quite clear here.

Also, I'm wondering if the Type 1 description here means that we shouldn't use say the rear-facing camera on the mobile device. I guess what I'm thinking is that the front-facing camera (and likely any sensor) is probably of a lower resolution than the rear-facing camera, so if we are going to limit to just 2 cameras, should we limit to the front-facing mobile device camera or should it be the "best" mobile device camera that is at least as good as the sensor?

So for example:

MD Specs = 8MP resolution, FF camera = 10MP resolution, RF camera = 16MP resolution

So should we limit Type 1 to the FF camera, or use the RF camera? SNS is likely to be from the FF camera (at least in terms of selfies, obviously), but there may be a lot of other pictures from someone's RF camera (spouse, friends, etc) with similar devices. 

I guess my real question is on the Type 1 and how we think it best to specify. I'm OK with Type 2 being a dSLR with let's say a minimum resolution of 3X the sensor and released within 2 years of the testing (i.e. no older than roughly 2 years old). I don't know that I would want to specify a specific resolution, unless we go with something like 3X or 40MP, whichever is greater (so you would end up with at least a 40MP camera, but if the sensor had a really high resolution, you could end up needing to get something better).

**Comment by n-kai** [comment]
I recommend to use a) at least two different types of camera and b) higher resolution PAIs based on 2D face research papers so how about the following specification?

Type 1: Rear-facing camera of a smartphone. This smartphone's sensor has the similar resolution of the TOE's one and released within 2 years of the testing.  
Type 2: dSLR camera. Camera's resolution should at least 2 times higher than the TOE's sensor and released within 2 years of the testing. 

I just want to avoid to specify any specific value (e.g. 40MP) because we have to change those values as the technology progresses.

**Comment by woodbe** [comment]
Closing as the current definitions are related to the device as the Type 1 and the Type 2 being based on a "new" camera in market. This has been handled in the subsequent toolboxes.
